### PR TITLE
MOAR

### PR DESCRIPTION
**Pull request by stefanpenner**
- [x] encode/decode executables
- [x] encoder
- [x] improve encode/decoding symmetry.
- [ ] concat
  - [ ] make concat handle all those edge cases
  - [ ] make fast
  - [ ] concat util
- [ ] make tests run in the browser again

### CODE REVIEW

**Comment by krisselden** [comment]
Can this be merged and the remaining stuff separate issues?

**Comment by stefanpenner** [comment]
Ya sure, this is largely just for you (or others to view and provide feedback on)

**Comment by krisselden** [comment]
Does this assume that no newline will be added between the concatenated generated sources?

**Comment by krisselden** [comment]
Also, there are cases where sloppy sourcemaps do not include all of the trailing lines after source code `;;;` also have to be careful about removing the source mapping url.

**Comment by stefanpenner** [comment]
@krisselden the concatenator is not complete. Basically just a naive first pass to allow us to produce the simplest of end to end tests. 

Follow up work intends to add addition scenarios (like the ones you pointed out) and benchmarks.

**Comment by Turbo87** [comment]
@stefanpenner what's the state of this PR and the repo in general? any low hanging fruits to move this forward?

**Comment by stefanpenner** [comment]
@Turbo87 the PR basically finishes up some of the fast-source-map machinery to support concatenation, but it does not complete the concatenation work.

If you are interested in helping, we can plan for a hangout this week. :)

**Comment by krisselden** [comment]
@stefanpenner can you cleanup the history? I don't want to just squash it all but some of the commits like moar would be nice to cleanup. I worry about long lived branches

**Comment by stefanpenner** [comment]
@krisselden not right now, maybe later this week

File: index.js [hunk 1]
```diff
@@ -181,47 +197,272 @@
...
+              srcCol: mapping.srcCol
+            };
+          });
+
+          return {
+            mappings: transformedLineMappings
+          };
+        }));
+
+        offset += 1;
```
Comment (Turbo87) [inline index.js]: shouldn't this be something like `offset += map.sources.length`?

Comment (stefanpenner) [inline index.js]: more then likely

**Comment by Turbo87** [comment]
I've cleaned up the branch/history: https://github.com/krisselden/fast-source-map/compare/master...Turbo87:concatenator-cleanup

**Comment by stefanpenner** [comment]
@Turbo87 looks very good, I have updated with your cleaned commit yesterday. Thank you very much!

**Comment by stefanpenner** [comment]
> @stefanpenner can you cleanup the history?

thanks to @Turbo87 this has now been completed

**Comment by Turbo87** [comment]
@krisselden this should be ready for merging now. please let me know if you need anything else to be done here that's blocking the merge.

**Comment by Turbo87** [comment]
@krisselden ping :)
